src/python/bot/tasks/fuzz_task.py

```diff
@@ -1254,319 +1258,391 @@ def get_strategy_distribution_from_ndb():
-  # variable so we can access it in launcher.
-  if environment.get_value('USE_BANDIT_STRATEGY_SELECTION'):
-    distribution = get_strategy_distribution_from_ndb()
-    environment.set_value('STRATEGY_SELECTION_DISTRIBUTION', distribution)
-
-  # Reset memory tool options.
-  environment.reset_current_memory_tool_options(redzone_size=redzone)
-
-  thread_wait_timeout = 1
-
-  # Create a dict to store metadata specific to each testcase.
-  testcases_metadata = {}
-  for testcase_file_path in testcase_file_paths:
-    testcases_metadata[testcase_file_path] = {}
-
-    # Pick up a gesture to run on the testcase.
-    testcases_metadata[testcase_file_path]['gestures'] = pick_gestures(
-        test_timeout)
-
-  logs.log('Starting to process testcases.')
-  logs.log('Redzone is %d bytes.' % redzone)
-  logs.log('Timeout multiplier is %s.' % str(timeout_multiplier))
-  logs.log('App launch command is %s.' %
-           testcase_manager.get_command_line_for_application())
-
-  # Start processing the testcases.
-  while test_number < len(testcase_file_paths):
-    thread_index = 0
-    threads = []
-
-    temp_queue = process_handler.get_queue()
-    if not temp_queue:
-      process_handler.terminate_stale_application_instances()
-      logs.log_error('Unable to create temporary crash queue.')
-      break
-
-    while thread_index < max_threads and test_number < len(testcase_file_paths):
-      testcase_file_path = testcase_file_paths[test_number]
-      gestures = testcases_metadata[testcase_file_path]['gestures']
-
-      env_copy = environment.copy()
-      thread = process_handler.get_process()(
-          target=testcase_manager.run_testcase_and_return_result_in_queue,
-          args=(temp_queue, thread_index, testcase_file_path, gestures,
-                env_copy, True))
-
-      try:
-        thread.start()
-      except:
+  def do_blackbox_fuzzing(self, fuzzer, fuzzer_directory, job_type):
+    """Run blackbox fuzzing."""
+    # Set the thread timeout values.
+    # TODO(ochang): Remove this hack once engine fuzzing refactor is compelte.
+    fuzz_test_timeout = environment.get_value('FUZZ_TEST_TIMEOUT')
+    if fuzz_test_timeout:
+      test_timeout = set_test_timeout(fuzz_test_timeout,
+                                      self.timeout_multiplier)
+    else:
+      test_timeout = self.test_timeout
+
+    thread_timeout = test_timeout
+
+    # Determine number of testcases to process.
+    testcase_count = environment.get_value('MAX_TESTCASES')
+
+    # For timeout multipler greater than 1, we need to decrease testcase count
+    # to prevent exceeding task lease time.
+    if self.timeout_multiplier > 1:
+      testcase_count /= self.timeout_multiplier
+
+    # Run the fuzzer to generate testcases. If error occurred while trying
+    # to run the fuzzer, bail out.
+    (error_occurred, testcase_file_paths,
+     sync_corpus_directory, fuzzer_metadata) = run_fuzzer(
```

Comment:
I wonder if run_fuzzer would be better as a method.

Reply:
Ack. There are many things which can be moved, will be done in future CLs to avoid making the diffs too large.